rplugin/python3/nvim-typescript/__init__.py

```diff
@@ -213,24 +220,33 @@ def tssig(self):
                 'echohl WarningMsg | echo "TS: Server is not Running" | echohl None')
 
     # Various Auto Commands
+    @neovim.autocmd('CursorHold', pattern='*.ts,*.tsx,*.js,*.jsx')
+    def on_cursorhold(self):
+        """
+        get type info on cursor hold
+        """
+        self.vim.command('TSType')
 
-    @neovim.autocmd('BufEnter', pattern='*.ts,*.tsx')
+    @neovim.autocmd('BufEnter', pattern='*.ts,*.tsx,*.js,*.jsx')
     def on_bufenter(self):
         """
            Send open event when a ts file is open
 
         """
-        if self.server is None:
-            self._client.start()
-            self.server = True
-            self.vim.out_write('TS: Server Started \n')
-            self._client.open(self.relative_file())
-        else:
-            self._client.open(self.relative_file())
+        if self.is_valid(self.relative_file()):
```

Comment:
with `self.is_valid` maybe instead we should do some os.walk instead. On vacation right now, sorry this is taking longer than expected. I'll post something that I'm toying with in a few days